governator-core/src/main/java/com/netflix/governator/internal/PreDestroyLifecycleActions.java

```diff
@@ -2,55 +2,103 @@
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 import javax.annotation.PreDestroy;
 
 import com.netflix.governator.LifecycleAction;
 
 /**
- * Special AbstractLifecycleFeature to support @PreDestroy annotation processing.
- * Note that this feature is implicit in LifecycleModule and therefore does not need 
- * to be added using the LifecycleFeature multibinding.
+ * Special AbstractLifecycleFeature to support @PreDestroy annotation processing and
+ * java.io.AutoCloseable detection. Note that this feature is implicit in LifecycleModule 
+ * and therefore does not need to be added using the LifecycleFeature multibinding.
  * 
  * @author elandau
  */
-public final class PreDestroyLifecycleActions extends AbstractLifecycleFeature {
+public final class PreDestroyLifecycleActions extends AbstractLifecycleFeature<Set<String>> {
```

Comment:
This seems like too much 'flexibility' and would result in a confusing API.  Why not just make so that getMethodActions takes a Set<Method> and getFieldActions task a Set<Field>?  Actually, what would be the use case for field actions?